src/mds/Server.cc

```diff
@@ -503,6 +509,14 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
     return;
   }
 
+  std::string_view fs_name = mds->get_fs_name();
+  if (!fs_name.empty() && !session->fs_name_capable(fs_name, MAY_READ)) {
+    dout(0) << " dropping message not allowed for this fs_name: " << *m << dendl;
+    auto reply = make_message<MClientSession>(CEPH_SESSION_REJECT);
```

Comment:
add reply metadata as in the sessionless case above

Reply:
Something like this - `reply->metadata["error_string"] = "client doesn't have caps for fsid "`? I did add it before but had to remove it since it broke backward compatibility. See - https://github.com/ceph/ceph/pull/32581#discussion_r415797264.

Reply:
Ya I'm confused by Zheng's comment. We do that for other reject messages already. @ukernel ?

Reply:
@batrick In case this helps, here's the related BZ - https://bugzilla.redhat.com/show_bug.cgi?id=1828340

Reply:
call send_reject_message(), it checks if client can handle error string

Reply:
Added the message again.